src/network/networkd-link.c

```diff
@@ -994,16 +994,17 @@ static int link_push_uplink_dns_to_dhcp_server(Link *link, sd_dhcp_server *s) {
 
         if (link->network->dhcp_use_dns && link->dhcp_lease) {
                 const struct in_addr *da = NULL;
-                int n;
+                int j, n;
```

Comment:
i figure this should be size_t actually, after all this is about memory...

Reply:
in this case, the for loop goes over `j < n`, and `n` is returned by `sd_dhcp_lease_get_dns()`.

For better or worse, `sd_dhcp_lease_get_dns()` returns an integer to leave room for error codes.

Mixing types here like doesn't seem preferable to me, is it?